src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToPackAToolProject.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+using NuGet.Packaging;
+
+namespace Microsoft.NET.ToolPack.Tests
+{
+    public class GivenThatWeWantToPackAToolProject : SdkTest
```

Comment:
Additional test scenarios I recommend covering:

- Multi-targeted projects (ie `<TargetFrameworks>netcoreapp2.1;netcoreapp2.0</TargetFrameworks>`)
- Tools which depend on a graph of NuGet packages (IE tool depends on package A which depends on package B and C, and verify that A.dll, B.dll, and C.dll are all packaged with the tool)
- Tools which depend on NuGet packages that have assets (ie native assets) that get deployed to a subfolder.  A sample package to use for this could be `Microsoft.DiaSymReader.Native`

Reply:
Fixed as what we discuss on live share

Reply:
And I actually find a bug of subfolder. I changed here https://github.com/dotnet/sdk/pull/1843/files#diff-c9ebf740daee06e626e6636288feb1d1R42 to preserve folder structure like tools/netcoreapp2.0/any/runtime/stuff